hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeParentRecordHandler.java

```diff
@@ -37,24 +38,46 @@
 import com.hedera.node.app.service.token.impl.WritableAccountStore;
 import com.hedera.node.app.service.token.impl.WritableNftStore;
 import com.hedera.node.app.service.token.impl.WritableTokenRelationStore;
+import com.hedera.node.app.service.token.impl.handlers.staking.StakingRewardsHandler;
 import com.hedera.node.app.service.token.impl.records.CryptoTransferRecordBuilder;
 import com.hedera.node.app.spi.workflows.HandleContext;
 import com.hedera.node.app.spi.workflows.HandleException;
 import com.hedera.node.app.spi.workflows.PreCheckException;
 import com.hedera.node.app.spi.workflows.PreHandleContext;
 import com.hedera.node.app.spi.workflows.TransactionHandler;
+import com.hedera.node.config.data.StakingConfig;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import javax.inject.Inject;
 import javax.inject.Singleton;
 
 /**
- * This is a special handler that is used to "finalize" hbar and token transfers, most importantly by writing the results of a transaction to record files. Finalization in this context means summing the net changes to make to each account's hbar balance and token balances, and assigning the final owner of an nft after an arbitrary number of ownership changes
+ * This is a special handler that is used to "finalize" hbar and token transfers for the parent transaction record.
+ * Finalization in this context means summing the net changes to make to each account's hbar balance and token
+ * balances, and assigning the final owner of an nft after an arbitrary number of ownership changes.
```

Comment:
Could reference https://github.com/hashgraph/hedera-services/issues/7084 here, noting that the modularized record of an NFT that goes through a "transfer chain" like `A -> B -> C` will look different than the mono-service record.

Reply:
yes Added.